spec/requests/search_controller_spec.rb

```diff
@@ -217,6 +267,57 @@
       expect(response.status).to eq(200)
       expect(SearchLog.where(term: 'bantha')).to be_blank
     end
+
+    context 'rate limited' do
+      it 'rate limits searches' do
+        RateLimiter.enable
+        RateLimiter.clear_all!
+
+        15.times do
+          get "/search.json", params: {
+                q: 'bantha'
+              }
+
+          expect(response.status).to eq(200)
+          json = response.parsed_body
+          expect(json["error"]).to eq(nil)
+        end
+
+        get "/search.json", params: {
+              q: 'bantha'
+            }
+        expect(response.status).to eq(200)
+        json = response.parsed_body
+        expect(json["grouped_search_result"]["error"]).to eq(I18n.t("rate_limiter.slow_down"))
+
+      end
+
+      context "and a logged in user" do
+        before { sign_in(Fabricate(:user)) }
+
+        it 'rate limits searches' do
+          RateLimiter.enable
+          RateLimiter.clear_all!
+
+          30.times do
```

Comment:
Just curious here - how slow are these tests? `30.times` seems like a lot.

Reply:
It was. I added a hidden site setting for this, and dialed the rate limits down in them so tests are much faster now